### ISSUE TITLE
gdb test hang on 

### ISSUE THREAD

**Issue by liuyd96** [avocado-framework/avocado#3521]
Hi, all. Recently i test avocado, gdb related test always hang on.
```
root@lkp-ivb-d01 /lkp/benchmarks/avocado/examples/tests# avocado run gdbtest.py:GdbTest.test_start_exit
JOB ID     : eb1033954a12e92bb4d02f1aff4ba2f58bc8df73
JOB LOG    : /tmp/avocado/result/job-2020-01-21T16.02-eb10339/job.log
 (1/1) gdbtest.py:GdbTest.test_start_exit: /^C
Interrupt requested. Waiting 2 seconds for test to finish (ignoring new Ctrl+C until then)
-
```
It seems something wrong with python module gdb (avocado/utils/gdb.py)
I write a test script to call gdb module, each time i interrupt test, show below info. 
```
root@lkp-ivb-d01 ~# ./test.py 
i: 0
^CTraceback (most recent call last):
  File "./test.py", line 14, in <module>
    test()
  File "./test.py", line 12, in test
    gdb_instances.append(gdb.GDB())
  File "/usr/local/lib/python3.5/dist-packages/avocado/utils/gdb.py", line 345, in __init__
    self.read_until_break()
  File "/usr/local/lib/python3.5/dist-packages/avocado/utils/gdb.py", line 397, in read_until_break
    line = self.read_gdb_response()
  File "/usr/local/lib/python3.5/dist-packages/avocado/utils/gdb.py", line 384, in read_gdb_response
    time.sleep(timeout)
KeyboardInterrupt
```
I checked avocado/utils/gdb.py, each time i run test.py, script fall into 'time.sleep(timeout).
Test env. 
```
root@lkp-ivb-d01 ~# python3 -V
Python 3.5.3
root@lkp-ivb-d01 ~# uname -a
Linux lkp-ivb-d01 5.5.0-rc5 #1 SMP Mon Jan 6 08:41:17 CST 2020 x86_64 GNU/Linux

```

**Comment by willianrampazzo** [comment]
This may be related to #3504.

**Comment by clebergnu** [comment]
Fixed by #3856
